### PR TITLE
Ensure users with only 'Reader' on dossier see closed tasks

### LINKED ISSUE #4887: Reader is not able to see closed tasks inside a dossier.
Users which only have the Reader role on a dossier are not able to see closed tasks. In my Opinion thats wrong and is now a real problem since we remove local roles on task when closing them.

Example: https://dev.onegovgever.ch/fd/ordnungssystem/fuehrung/gemeinderecht/uebergeordnete-erlasse/dossier-448#overview

See [extranet issue](https://extranet.4teamwork.ch/support/gemeinde-koeniz/tracker/102)

### PR DESCRIPTION
**Pull request by Rotonen**
Closes #4887

### CODE REVIEW

**Comment by Rotonen** [comment]
Got it now, but I guess we actually would like to acquire ´View` wider than just for the closed state?

**Comment by Rotonen** [comment]
@phgross so, maybe I finally got what's going on - amended what the local roles setter does for tasks which are in the final state.

File: opengever/task/localroles.py [hunk 1]
```diff
@@ -23,14 +23,17 @@ def __init__(self, task):
...
     def set_roles(self, event):
-        if self.task.is_in_final_state:
-            return
-
         self.event = event
-        self.set_roles_on_task()
-        self.globalindex_reindex_task()
-        self.set_roles_on_related_items()
-        self.set_roles_on_distinct_parent()
+
```
Comment (phgross) [inline opengever/task/localroles.py]: No, thats wrong, you add access to responsible users, which is not the case of the issue/pr and definitely not the wish (with OGIP 45 we recently introduced the exact opposite of this change - the revoke of local roles on task closing).

The only change which is needed is to ensure that users with only the 'Reader' role on a dossier are able to see closed tasks:
 => Add `Reader` role to the `view` permission for closed task state, in the task workflow.

**Comment by Rotonen** [comment]
@phgross do I:

1) add to the forbiddenfruit mess of our unit tests to make `api.user.has_permission()` work in the unit tests
2) port the task transition controller unit tests to integration tests

?

**Comment by Rotonen** [comment]
@phgross ready for rereview.

File: opengever/testing/builders/dx.py [hunk 2]
```diff
@@ -228,6 +228,7 @@ def successor_from(self, predecessor):
 
     def change_workflow_state(self, obj):
         super(TaskBuilder, self).change_workflow_state(obj)
+        obj.get_sql_object().review_state = api.content.get_state(obj)
```
Comment (phgross) [inline opengever/testing/builders/dx.py]: Thats not necessary, the sync_task(obj, None) a line below, those also update the review_state.

File: opengever/task/tests/test_localroles.py [hunk 3]
```diff
@@ -198,6 +198,29 @@ def test_closing_a_task_revokes_responsible_roles_on_task(self, browser):
...
+        self.login(self.reader_user, browser)
+        browser.open(self.task, view='tabbedview_view-overview')
+        expected_actions = ['task-transition-delegate', 'task-transition-reassign']
+        self.assertEqual(expected_actions, browser.css('#action-menu a').text)
+
+        with self.login(self.dossier_responsible, browser=browser):
+            self.set_workflow_state('task-state-tested-and-closed', self.subtask)
+            self.set_workflow_state('task-state-resolved', self.task)
+
+        browser.open(self.task, view='tabbedview_view-overview')
```
Comment (phgross) [inline opengever/task/tests/test_localroles.py]: Hmm you test with `dossier_responsible`, but I would say he got more than the `Reader` role.

File: opengever/testing/builders/dx.py (hunk 2 above)
Comment (Rotonen) [reply opengever/testing/builders/dx.py]: IIRC that only did it for the Plone object, thus I ended up doing this, but will reverify.

File: opengever/task/tests/test_localroles.py (hunk 3 above)
Comment (Rotonen) [reply opengever/task/tests/test_localroles.py]: ?

File: opengever/testing/builders/dx.py (hunk 2 above)
Comment (Rotonen) [reply opengever/testing/builders/dx.py]: Yep, seems I missed the `globalindex/model` logic and forgot to requery in early testing. Dropping this commit.

**Comment by Rotonen** [comment]
@phgross ready for rereview.

File: opengever/task/tests/test_localroles.py (hunk 3 above)
Comment (phgross) [reply opengever/task/tests/test_localroles.py]: Too early review 🙈 ...

**Pull request merged**